hr_expense_advance_clearing/models/hr_expense_sheet.py

```diff
@@ -138,15 +138,22 @@ def open_clear_advance(self):
         vals["context"] = context1
         return vals
 
+    def get_domain_advance_sheet_expense_line(self):
+        return self.advance_sheet_id.expense_line_ids.filtered("clearing_product_id")
+
+    def create_clearing_expense_line(self, line):
+        clear_advance = self._prepare_clear_advance(line)
+        clearing_line = self.env["hr.expense"].new(clear_advance)
+        return clearing_line
+
     @api.onchange("advance_sheet_id")
     def _onchange_advance_sheet_id(self):
         self.expense_line_ids -= self.expense_line_ids.filtered("av_line_id")
         self.advance_sheet_id.expense_line_ids.sudo().read()  # prefetch
-        for line in self.advance_sheet_id.expense_line_ids.filtered(
-            "clearing_product_id"
-        ):
-            clear_advance = self._prepare_clear_advance(line)
-            self.expense_line_ids += self.env["hr.expense"].new(clear_advance)
+        lines = self.get_domain_advance_sheet_expense_line()
+        self.expense_line_ids = [
```

Comment:
Not `+=` ?

Reply:
The original code, what it try to do is to add lines, not assign lines, isn't it?
(says, user can also manually add some line first, before trigger onchange)

Reply:
Okay, I forget this case.
Fixed.

Thank you for your reviewed.